pkg/fetch/windows/csaf/types.go

```diff
@@ -0,0 +1,226 @@
+	CVE             string           `json:"cve,omitempty"`
+	CWE             *struct {
+		ID   string `json:"id"` // pattern: ^CWE-[1-9]\\d{0,5}$
+		Name string `json:"name"`
+	} `json:"cwe,omitempty"`
+	DiscoveryDate string `json:"discovery_date,omitempty"`
+	Flags         []struct {
+		Date       string           `json:"date,omitempty"`
+		GroupIDs   []ProductGroupID `json:"group_ids,omitempty"`
+		Label      string           `json:"label"` // enum: component_not_present, inline_mitigations_already_exist, vulnerable_code_cannot_be_controlled_by_adversary, vulnerable_code_not_in_execute_path, vulnerable_code_not_present
+		ProductIDs []ProductID      `json:"product_ids,omitempty"`
+	} `json:"flags,omitempty"`
+	IDs []struct {
+		SystemName string `json:"system_name"`
+		Text       string `json:"text"`
+	} `json:"ids,omitempty"`
+	Involvements []struct {
+		Date    string `json:"date,omitempty"`
+		Party   string `json:"party"`  // enum: coordinator, discoverer, other, user, vendor
+		Status  string `json:"status"` // enum: completed, contact_attempted, disputed, in_progress, not_contacted, open
+		Summary string `json:"summary,omitempty"`
+	} `json:"involvements,omitempty"`
+	Notes         []Note `json:"notes,omitempty"`
+	ProductStatus struct {
+		FirstAffected      []ProductID `json:"first_affected,omitempty"`
+		FirstFixed         []ProductID `json:"first_fixed,omitempty"`
+		Fixed              []ProductID `json:"fixed,omitempty"`
+		KnownAffected      []ProductID `json:"known_affected,omitempty"`
+		KnownNotAffected   []ProductID `json:"known_not_affected,omitempty"`
+		LastAffected       []ProductID `json:"last_affected,omitempty"`
+		Recommended        []ProductID `json:"recommended,omitempty"`
+		UnderInvestigation []ProductID `json:"under_investigation,omitempty"`
+	} `json:"product_status,omitempty"`
+	References []struct {
+		Category string `json:"category,omitempty"`
+		Summary  string `json:"summary,omitempty"`
+		URL      string `json:"url,omitempty"`
+	} `json:"references,omitempty"`
+	ReleaseDate  string `json:"release_date,omitempty"`
+	Remediations []struct {
+		Category        string           `json:"category"` // enum: mitigation, no_fix_planned, none_available, vendor_fix, workaround
+		Date            string           `json:"date,omitempty"`
+		Details         string           `json:"details"`
+		Entitlements    []string         `json:"entitlements,omitempty"`
+		GroupIDs        []ProductGroupID `json:"group_ids,omitempty"`
+		ProductIDs      []ProductID      `json:"product_ids,omitempty"`
+		RestartRequired *struct {
+			Category string `json:"category"` // enum: connected, dependencies, machine, none, parent, service, system, vulnerable_component, zone
+			Details  string `json:"details,omitempty"`
+		} `json:"restart_required,omitempty"`
+		URL string `json:"url,omitempty"`
+	} `json:"remediations,omitempty"`
+	Scores []struct {
+		CvssV2 *struct {
+			AccessComplexity      string  `json:"accessComplexity,omitempty"`
```

Comment:
Use camelCase here? Other json keys seems usually underscore_connected.

Reply:
The csaf format uses camel case, so it is adapted to that.
https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html

Reply:
Understood! Good choice!